dns.go

```diff
@@ -0,0 +1,113 @@
+package gubernator
+
+import (
+	"context"
+	"time"
+
+	"github.com/mailgun/holster/v4/setter"
+	"github.com/miekg/dns"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type DNSPoolConfig struct {
+	// (Required) The FQDN that should resolve to gubernator instance ip addresses
+	FQDN string
+
+	// (Required) Own GRPC address
+	OwnAddress string
+
+	// (Required) Called when the list of gubernators in the pool updates
+	OnUpdate UpdateFunc
+
+	Logger logrus.FieldLogger
+}
+
+type DNSPool struct {
+	log    logrus.FieldLogger
+	conf   DNSPoolConfig
+	ctx    context.Context
+	cancel context.CancelFunc
+}
+
+func NewDNSPool(conf DNSPoolConfig) (*DNSPool, error) {
+	setter.SetDefault(&conf.Logger, logrus.WithField("category", "gubernator"))
+
+	if conf.OwnAddress == "" {
+		return nil, errors.New("Advertise.GRPCAddress is required")
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	pool := &DNSPool{
+		log:    conf.Logger,
+		conf:   conf,
+		ctx:    ctx,
+		cancel: cancel,
+	}
+	go pool.task()
+	return pool, nil
+}
+
+func peer(ip string, self string, ipv6 bool) PeerInfo {
+
+	if ipv6 {
+		ip = "[" + ip + "]"
+	}
+	grpc := ip + ":81"
+	return PeerInfo{
+		DataCenter:  "",
+		HTTPAddress: ip + ":80",
+		GRPCAddress: grpc,
+		IsOwner:     grpc == self,
+	}
+
+}
+
+func (x *DNSPool) task() {
+	for {
+		var delay uint32 = 10
+		config, _ := dns.ClientConfigFromFile("/etc/resolv.conf")
```

Comment:
Might want to make this an optional config item.

Reply:
I'm not sure I understand, what would be optional or configurable, the name server to use or something else?

Reply:
The path for the `"/etc/resolv.conf"` should be optional, but have a default of `"/etc/resolv.conf"`

Reply:
Like... for testing?
I can't think of any environment where resolver configuration file would reside at a different path...
(There's Windows ofc., but that doesn't even use a file)